packages/types/src/Metadata/v1/index.ts

```diff
@@ -0,0 +1,14 @@
+// Copyright 2017-2019 @polkadot/types authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import Struct from '../../codec/Struct';
+import MagicNumber from './MagicNumber';
+
+export default class MetadataV1 extends Struct {
+  constructor (value?: any) {
+    super({
+      magicNumber: MagicNumber
```

Comment:
Yeap... basically if you look at the current, it is basically a proxy parser - try one, then the other. All the types passed through are still for V0. What I will probably do as a first stab is to 

- get the parsing for both formats going (this is where the magic number is used to detect)
- define the V1 types
- have a conversion tool from V1 -> v0 - so basically we still use V0-structures everywhere

Then once we have these rolled out can remove support for V0 (`Null` in the enum), remove the `try/catch` on parsing (all these are one-liners) and then get all the storage, extrinsics, etc. to use the V1 format natively

The latter part, since I need to support both formats and need a conversion tool, will leave out from this PR - it will just add more moving parts.